presto-testing/src/main/java/io/prestosql/testing/AbstractTestQueryFramework.java

```diff
@@ -294,9 +293,28 @@ protected void assertAccessDenied(@Language("SQL") String sql, @Language("RegExp
     }
 
     protected void assertAccessDenied(
+            @Language("SQL") String sql,
+            @Language("RegExp") String exceptionsMessageRegExp,
+            @Language("RegExp") String exceptionsCauseMessageRegExp,
+            TestingPrivilege... deniedPrivileges)
+    {
+        assertAccessDenied(getSession(), sql, ".*" + exceptionsMessageRegExp + ": Access Denied: " + exceptionsCauseMessageRegExp, ".*Access Denied: " + exceptionsCauseMessageRegExp, deniedPrivileges);
+    }
+
+    protected void assertAccessDenied(
+            Session session,
+            @Language("SQL") String sql,
+            @Language("RegExp") String exceptionsMessageRegExp,
+            TestingPrivilege... deniedPrivileges)
+    {
+        assertAccessDenied(getSession(), sql, ".*Access Denied: " + exceptionsMessageRegExp, null, deniedPrivileges);
+    }
+
+    private void assertAccessDenied(
             Session session,
             @Language("SQL") String sql,
             @Language("RegExp") String exceptionsMessageRegExp,
+            @Language("RegExp") String exceptionsCauseMessageRegExp,
```

Comment:
i'd suggest decoupling running query with verifying the exception using 
`org.assertj.core.api.Assertions#assertThatThrownBy(org.assertj.core.api.ThrowableAssert.ThrowingCallable)`

a helper like `io.prestosql.testing.assertions.PrestoExceptionAssert#assertPrestoExceptionThrownBy` can be useful

Reply:
done